lib/src/printers/pretty_printer.dart

```diff
@@ -140,21 +198,110 @@ class PrettyPrinter extends LogPrinter {
     }
   }
 
-  bool _discardDeviceStacktraceLine(String line) {
+  final Map<Level, Set<String>> _ignoredPackagesByLevel = {} ;
+  final Set<String> _ignoredPackages = {} ;
+
+  /// Returns a list of ignored packages while printing the stack trace.
+  List<String> get ignoredPackages {
+    var l1 = _ignoredPackagesByLevel.values.expand((e) => e) ;
+    var set = Set.from( l1 ) ;
+    set.addAll( _ignoredPackages ) ;
+    return List.from(set) ;
+  }
+
+  /// Clears the ignored packages.
+  void clearIgnoredPackages() {
+    _ignoredPackagesByLevel.clear();
+    _ignoredPackages.clear() ;
+  }
+
+  /// Ignores a [package] in stack trace printing.
+  ///
+  /// [level] If is provided, [package] will be ignored only for this [level].
+  void ignorePackage(String package, [Level level]) {
+    if (package == null) return ;
+    package = package.trim() ;
+    if (package.isEmpty) return ;
+
+    if (level == null) {
+      _ignoredPackages.add(package) ;
+    }
+    else {
+      var set = _ignoredPackagesByLevel.putIfAbsent(level, () => {}) ;
+      set.add(package) ;
+    }
+  }
+
+  /// Removes a [package] from ignore list.
+  ///
+  /// [level] If provided will remove it only if defined for this [Level].
+  bool doNotIgnorePackage(String package, [Level level]) {
```

Comment:
Just return `void` here. 

I'm actually on the fence with regard to this functionality. Perhaps a new printer would be better.

Reply:
PrettyPrinter is already handling packages to ignore (formatStackTrace). Maybe in the future this functionality can be moved to another class that can be used by any printer.